src/BenchmarkDotNet/ListBenchmarks/TreeBenchmarkCasesPrinter.cs

```diff
@@ -0,0 +1,39 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+
+namespace BenchmarkDotNet.ListBenchmarks {
+    internal  class TreeBenchmarkCasesPrinter : IBenchmarkCasesPrinter
+    {
+        public void Print(IEnumerable<string> testName)
+        {
+            List<Node> topLevelNodes = new List<Node>();
+
+            foreach (string test in testName)
+            {
+                var partsOfName = test.Split('.');
+                PrepareNodeTree(topLevelNodes, partsOfName);
+            }
+
+            var tree = new AsciiTreeDiagram();
```

Comment:
personally, I would merge the body of `AsciiTreeDiagram` into this class. Both have the same responsibility - printing a tree. We could keep `AsciiTreeDiagram` if it had some unit tests